src/v/raft/persisted_stm.h

```diff
@@ -160,6 +160,13 @@ concept supported_stm_snapshot = requires(T s, stm_snapshot&& snapshot) {
  *
  * To speed up the catch up process persisted_stm snapshots the state
  * and uses it as a base for replaying the commands.
+ *
+ * !!!! WARNING !!!!
+ * Removal of the persisted stm local snapshot.
+ *
+ * Removing persistent stm snapshot will trigger its initial recovery policy to
+ * be applied. This may lead to skipping some of the batches that may be
+ * important when the stm uses `skip_to_end` recovery policy.
```

Comment:
I don't have a good/strong suggestion here, but it might be nice if we had a place to store the existence of STMs (like some kind of stm manager snapshot). Then we could tell if this were indeed the first time bootstrapping or if the local snapshot were deleted.

Reply:
I actually pursued this idea, and change the way initial recovery information is stored